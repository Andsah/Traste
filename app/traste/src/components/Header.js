import React from "react";
import AppBar from "@mui/material/AppBar";
import Box from "@mui/material/Box";
import Toolbar from "@mui/material/Toolbar";
import Typography from "@mui/material/Typography";

import logo from "../assets/traste_logo.png";
import KeyboardBackspaceIcon from "@mui/icons-material/KeyboardBackspace";
import {Colors} from "../assets/Colors"


function Header(props) {
  return (
    <Box data-testid="header">
      <AppBar
        position="static"
        sx={{ textAlign: "center", backgroundColor: Colors.trasteNavyBlue }}
      >
        <Toolbar>
          <KeyboardBackspaceIcon fontSize="large" onClick={props.goBackHandler} />

          <Typography variant="h6" component="div" sx={{ flexGrow: 1 }}>
            TRASTE
          </Typography>
          <img src={logo} alt="traste logo" style={{ height: "40px" }} />
        </Toolbar>
      </AppBar>
    </Box>
  );
}
export default Header;
