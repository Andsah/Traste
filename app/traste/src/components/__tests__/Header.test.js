import { render, screen, cleanup } from "@testing-library/react";
import Header from "../Header";
import renderer from "react-test-renderer";

afterEach(() => {
  cleanup();
});

test("Should contain TRASTE text", () => {
  render(<Header />);
  const headerElement = screen.getByTestId("header");
  expect(headerElement).toBeInTheDocument();
  expect(headerElement).toHaveTextContent("TRASTE");
});

test("Should not contain SIMON text", () => {
  render(<Header />);
  const headerElement = screen.getByTestId("header");
  expect(headerElement).toBeInTheDocument();
  expect(headerElement).not.toHaveTextContent("SIMON");
});

test("Matches snapshot", () => {
  const tree = renderer.create(<Header />).toJSON();
  expect(tree).toMatchSnapshot();
});
