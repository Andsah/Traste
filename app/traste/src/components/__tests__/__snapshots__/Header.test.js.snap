// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<div
  className="MuiBox-root css-0"
  data-testid="header"
>
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-ssakjc-MuiPaper-root-MuiAppBar-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
        data-testid="KeyboardBackspaceIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M21 11H6.83l3.58-3.59L9 6l-6 6 6 6 1.41-1.41L6.83 13H21z"
        />
      </svg>
      <div
        className="MuiTypography-root MuiTypography-h6 css-1juivf6-MuiTypography-root"
      >
        TRASTE
      </div>
      <img
        alt="traste logo"
        src="traste_logo.png"
        style={
          Object {
            "height": "40px",
          }
        }
      />
    </div>
  </header>
</div>
`;
